### PR TITLE
Intro docs 79

### PR DESCRIPTION
**Pull request by kay-kim**
#. Commit 1 adds a little intro to lay context for the libraries.  Any Solid-specific things are linked to the authoritative source.
https://docs-client-lib-js-git-intro-docs-79.inrupt.vercel.app/introduction/

#. Commit 2 adds a Getting Started Part 2 (which performs a creation operation). 
https://docs-client-lib-js-git-intro-docs-79.inrupt.vercel.app/tutorial/getting-started-part2/

~~As mentioned, the authn-browser package downloaded from npm has a dependency issue. To test, I actually had to pin the version as well as to pass in the option when creating a new Session({clientAuthentication: getClientAuthenticationWithDependencies({})});~~  Commit #4 should resolve this.

#. Commit 3 adds a little blurb to clarify between saveAt and saveInContainer.

#. Commit 4 adds a step to explicitly install our libraries to work around the npm installed libraries having the dependency error.

### CODE REVIEW

File: source/includes/table-client-libraries.rst [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
+     - A client library for accessing data stored in `Solid
+       <https://solidproject.org/>`_ :term:`Pods <Pod>`.
 
    * - ``solid-client-authn``
 
-     - Inrupt ``solid-client-authn`` is a set of libraries for
-       authenticating to Solid identity servers.
+     - A set of libraries for authenticating to Solid identity servers.
 
    * - ``vocab-common-rdf``
```
Comment (Vinnl) [inline source/includes/table-client-libraries.rst:17]: @nicolasmondada I'm not sure what the correct names for the vocabs are - I presume this, judging by the READMEs?

```suggestion
   * - The Solid RDF Vocab library for JavaScript
```

File: source/includes/table-client-libraries.rst [hunk 2]
```diff
@@ -28,12 +26,10 @@
 
    * - ``vocab-solid-common``
```
Comment (Vinnl) [inline source/includes/table-client-libraries.rst:27]:
```suggestion
   * - The Solid Common Vocab library for JavaScript
```

File: source/includes/table-client-libraries.rst [hunk 3]
```diff
@@ -28,12 +26,10 @@
...
 
-     - Inrupt ``vocab-solid-rdf`` is a library that provides
-       convenience objects for many
+     - A library that provides convenience objects for many
        Solid-related identifiers.
 
    * - ``vocab-inrupt-common``
```
Comment (Vinnl) [inline source/includes/table-client-libraries.rst:32]: @nicolasmondada I'm not sure what the correct names for the vocabs are - I presume this?

```suggestion
   * - The Solid Inrupt Vocab library for JavaScript
```

File: source/introduction.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+The JavaScript Client Libraries provide a suite of JavaScript
+Application Programming Interfaces (APIs) to build `Solid
+<https://solid.github.io/specification/>`__ applications.
+
+Solid Applications
+==================
+
+Solid applications access data stored in Solid :term:`Pods <Pod>`
+(Solid Pod is the storage location for data). Solid applications refer
+to applications that conform to the `Solid specification
```
Comment (Vinnl) [inline source/introduction.rst:14]: I recall heated arguments by Ruben insisting that we do not call Solid a specification, but rather IIRC "a set of specifications". :shrug:

File: source/introduction.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+<https://solid.github.io/specification/>`_ (similarly for Solid servers
+and Solid Pods).
+
+The specification defines various Solid application behavior, including
```
Comment (Vinnl) [inline source/introduction.rst:18]: Various implies plural?

```suggestion
The specification defines various Solid application behaviors, including
```

File: source/introduction.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+.. list-table::
+   :header-rows: 1
+
+   * - Browser
+
+     - Version
+
+   * - Google Chrome
+
+     - 83+
```
Comment (Vinnl) [inline source/introduction.rst:50]: This is already outdated, it appears:

```suggestion
     - 84+
```

File: source/introduction.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+   * - Mozilla Firefox
+
+     - 80+
+
+   * - Microsoft Edge
+
+     - 85+
```
Comment (Vinnl) [inline source/introduction.rst:58]: This, instead, is ahead on desktop:

```suggestion
     - 84+
```

Additionally, it appears that Edge's version numbers do not align across mobile and desktop, so perhaps we should list 43+ for mobile as well?

File: source/tutorial/getting-started-part2.rst [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+   The output resembles the following:
+
+   .. code-block:: none
+
+      Server running at http://localhost:1234
+      ...
+      Built in 2ms.
+
+   If there are warnings with regards to the various source maps, you
+   can ignore them.
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:162]: What errors are this? If it's something we can fix in the respective packages, that would be more professional than telling people to ignore warnings, IMHO.

File: source/tutorial/getting-started-part2.rst [hunk 9]
```diff
@@ -0,0 +1,414 @@
...
+      prompted to authorize ``http://localhost:1234`` to access your
+      Pod. To allow the application to read and write to your Pod,
+      click :guilabel:`Authorize`.
+
+   #. You are redirected back to your page.
+
+   .. tip::
+
+      - If you encounter redirect issues trying to log in, you may need
+        to clear the cache.
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:183]: :O @NSeydoux Is this still true, i.e. not a bug that was fixed recently?

File: source/tutorial/getting-started-part2.rst [hunk 10]
```diff
@@ -0,0 +1,414 @@
...
+      When redirected back from the identity provider, use
+      :apisolidauthnbrowser:`Session.handleIncomingRedirect()
+      </classes/_session_.session.html#handleincomingredirect>` to
+      complete the login process:
+
+      .. literalinclude:: /examples/getting-started/index2.js
+         :language: javascript
+         :lines: 57-75
+
+      The function collects the information provided by the identity
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:259]:
```suggestion
      The function verifies the information provided by the identity
```

@NSeydoux I think this might be more accurate?

File: source/tutorial/getting-started-part2.rst [hunk 11]
```diff
@@ -0,0 +1,414 @@
...
+      To the Thing, the application uses the following functions to add
+      data to the Thing:
+
+      - :apisolidclient:`addUrl </modules/_thing_add_.html#addurl>` to
+        add the ``RDF.type`` property with
+        ``http://schema.org/article`` as its value.
+
+      - :apisolidclient:`addStringNoLocale
+        </modules/_thing_add_.html#addstringnolocale>` to add the
+        ``SCHEMA_INRUPT_EXT.name`` property with the title as its value.
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:307]: Should we clarify why you need `SCHEMA_INRUPT_EXT`? (Because I have no idea...)

File: source/tutorial/getting-started-part2.rst [hunk 12]
```diff
@@ -0,0 +1,414 @@
...
+      Then, the application uses :apisolidclient:`setThing
+      </modules/_thing_thing_.html#setthing>` to add the Thing to the
+      SolidDataset (i.e., the reading list).
+
+      .. literalinclude:: /examples/getting-started/index2.js
+         :language: javascript
+         :lines: 88-94
+
+      The identifiers ``RDF.type``, and ``SCHEMA_INRUPT_EXT.name``, and
+      ``http://schema.org/article`` impose no restrictions on the data
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:318]: What's the difference between `http://schema.org/article` and the others, i.e. why is it listed as a URL whereas the others are listed as JS object constants?

File: source/tutorial/getting-started-part2.rst [hunk 13]
```diff
@@ -0,0 +1,414 @@
...
+      
+         The ``solid-client`` library also provides the
+         :apisolidclient:`saveSolidDatasetInContainer
+         </modules/_resource_soliddataset_.html#savesoliddatasetincontainer>`. 
+         However, unlike :apisolidclient:`saveSolidDatasetAt
+         </modules/_resource_soliddataset_.html#savesoliddatasetat>`
+         which creates any intermediate folders/containers as needed,
+         :apisolidclient:`saveSolidDatasetInContainer
+         </modules/_resource_soliddataset_.html#savesoliddatasetincontainer>`
+         requires the specified destination container already exists.
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:369]:
```suggestion
         requires that the specified destination container already exists.
```

(I think this is needed, but not a native speaker...)

File: source/tutorial/getting-started-part2.rst [hunk 14]
```diff
@@ -0,0 +1,414 @@
...
+
+   .. tab:: Verify
+
+      Upon save, the application re-retrieves the SolidDataset (the
+      reading list). The ``savedReadListDataSet`` returned from the
+      save operation reflects the saved state of the data. However, for
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:375]: This is not exactly true: it represents the data that was sent to the Pod. This distinction would primarily be relevant when updating an existing Resource: because someone with write access (such as the current user in a different tab/device) might have also written data to the same Resource in between fetching and saving the data, the actual data in the Pod might be different from what was sent to it, and doing another fetch afterwards will ensure that you have the latest-available data.

It's hard to judge how often that will be something to take into account when writing an app, i.e. how detailed we should describe that here, but I suppose we at least shouldn't be saying it's the _saved_ data:

```suggestion
      reading list). The ``savedReadListDataSet`` returned from the
      save operation reflects the sent data. However, for
```

(This is something that we ran into recently, so this wasn't clear to myself either until recently.)

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (Vinnl) [reply source/tutorial/getting-started-part2.rst:307]: (Specifically, the `_INRUPT_EXT` part would strike me as curious if I were following this tutorial.)

File: source/tutorial/getting-started-part2.rst [hunk 15]
```diff
@@ -0,0 +1,414 @@
...
+Additional Information
+======================
+
+API Documentation
+-----------------
+
+- `solid-client API
+  <https://docs.inrupt.com/developer-tools/api/javascript/solid-client/README.html>`__
+
+- `solid-client-authn-browser API
```
Comment (Vinnl) [inline source/tutorial/getting-started-part2.rst:405]: @nicolasmondada should we be using the other names here as well?

File: source/tutorial/getting-started.rst [hunk 16]
```diff
@@ -103,7 +105,17 @@ Build the Application
 
    b. When prompted ``Is this OK? (yes)``, enter to accept ``yes``.
 
-2. Install Parcel
+2. Install the Client Libraries
+-------------------------------
+
+#. Use ``npm`` to install the ``solid-client``,
+   ``solid-client-authn-browser``, and ``vocab-common-rdf`` libraries:
```
Comment (Vinnl) [inline source/tutorial/getting-started.rst:112]: @nicolasmondada and here?

File: source/tutorial/read-write-data.rst [hunk 17]
```diff
@@ -388,6 +390,18 @@ another process made separate non-conflicting modifications to the
 SolidDataset after you retrieved the SolidDataset but before you
 successfully saved, the returned SolidDataset reflects both the changes.
 
+.. [#behavior]
+      
+   The ``solid-client`` library also provides the
```
Comment (Vinnl) [inline source/tutorial/read-write-data.rst:395]:
```suggestion
   The Solid JavaScript Client library also provides the
```

**Review by Vinnl — APPROVED** [review]

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:375]: Oh -- so, we currently have on the read/write data

```
Upon successful save, saveSolidDatasetAt returns a SolidDatset that reflects its state after the save. That means, if another process made separate non-conflicting modifications to the SolidDataset after you retrieved the SolidDataset but before you successfully saved, the returned SolidDataset reflects both the changes
```
I thought I tested this and it did return both changes, but could be remembering incorrectly.

File: source/examples/getting-started/index2.js [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+// This calls the function to process login information.
+// If the function is called when not part of the login redirect, the function is a no-op.
+handleRedirectAfterLogin();
+
+// 2. Create the Reading List
+async function createList() {
+  labelCreateStatus.textContent = "";
+  const podURL = document.getElementById("PodURL").value;
+
+  const dsURL = `${podURL}/getting-started/readingList/myListDS`
```
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/examples/getting-started/index2.js:82]: Do you think adding a blurb that for simplicity, we're hardcoding. But in practice, you would have a predicate pointing to the location in their profile  would suffice?   (heh, of course, we're not showing people how to add the predicate but only so much one can do in a getting started w/o turning it into a novel 😄 )

File: source/includes/table-client-libraries.rst [hunk 19]
```diff
@@ -1,24 +1,22 @@
...
 
    * - ``solid-client``
 
-     - Inrupt ``solid-client`` is a client library for accessing data
-       stored in `Solid <https://solidproject.org/>`_ :term:`Pods
-       <Pod>`.
+     - A client library for accessing data stored in `Solid
+       <https://solidproject.org/>`_ :term:`Pods <Pod>`.
 
    * - ``solid-client-authn``
```
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/includes/table-client-libraries.rst:13]: Oh, so, if we change these in the Library column and we no longer mention the name of the actual library to download in the Description column, it'd be hard for users to find our libraries.  Do we put back the actual library name in the description column?
We could link to the npm site for the new names in the Library column, but that seems kind of bumpy to immediately get people away from our docs site when they're just getting started, yes?

File: source/introduction.rst (hunk 6 above)
Comment (kay-kim) [reply source/introduction.rst:50]: How funny. I did confirm these with @nicolasmondada on Oct 28 (https://docs.google.com/document/d/12831K81yC0B7bwAIVmetKXdAsAhRFtK8TBpiH1-g-LQ/edit).

So, I'll admit I'm a little confused about specifying minimum versions when we state "latest 2 stable releases".  In which case, do we need to mention the minimum versions (since they'll be changing at whatever rate they change).

File: source/includes/table-client-libraries.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/includes/table-client-libraries.rst:13]: Also, the reason I kind of like having more of the actual library name rather than the more generic product name  is because to have the name in people's mind when they're actually righting the code.

 import { ... } from "@inrupt/solid-client";

rather than having to do the association of Solid JavaScript Client -- and going, what was  the Solid JavaScript Authentication library's name (let me search this).  That is, I'm good with mentioning it like Solid JavaScript Client (``solid-client``)  here and there, but it seems like we'd want to mention ``solid-client`` more often in relations to pages with examples?

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (Vinnl) [reply source/tutorial/getting-started-part2.rst:375]: Huh... Unless I'm misreading the source code, I don't think that is the case. I'll do a test and report back.

File: source/examples/getting-started/index2.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (Vinnl) [reply source/examples/getting-started/index2.js:82]: From my point of view I think that's fine, but then I'm not a purist :)

File: source/includes/table-client-libraries.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (Vinnl) [reply source/includes/table-client-libraries.rst:13]: Yeah I agree. Nicolas just told me via chat that he didn't intend for these names to change in the library docs, so while I'd personally use the library names everywhere, it appears we can at least do so here :) I'll resolve the relevant conversations.

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:375]: Sounds good.  Will retest as well.

File: source/introduction.rst (hunk 6 above)
Comment (Vinnl) [reply source/introduction.rst:50]: 86 was released on October 6th according to https://chromereleases.googleblog.com/2020/10/stable-channel-update-for-desktop.html and https://en.wikipedia.org/wiki/Google_Chrome_version_history.

(I brought up the same point about the explicit versions as well - Chrome (and Edge, I think) releases every six weeks, Firefox every four.)

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (Vinnl) [reply source/tutorial/getting-started-part2.rst:375]: Just verified it, and indeed, the return value is not up-to-date with the saved Pod. [Test code.](https://codesandbox.io/s/optimistic-bose-f0ugm?file=/src/index.ts)

Screen recording of that same app running twice, with each instance editing different values in the same Resource:

concurrent-updates

Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:375]: You are absolutely correct.  I can't recreate what I thought I saw last time (where I was changing the Role at inrupt.net while an application was updating the name).  Will update this and the other blurb.

File: source/introduction.rst (hunk 4 above)
Comment (kay-kim) [reply source/introduction.rst:14]: So, not even mention Solid?    i.e., "that conform to a set of specifications"?

File: source/introduction.rst (hunk 6 above)
Comment (kay-kim) [reply source/introduction.rst:50]: Will update just to state "Latest 2 versions" in the versions column and use the browsers mentioned on 

https://github.com/inrupt/public-documentation/blob/ad6ded25a6340cc4e505de7fe6b7582d4586b96e/coding-conventions/javascript-coding-standards.md

File: source/introduction.rst (hunk 7 above)
Comment (kay-kim) [reply source/introduction.rst:58]: Adding browsers for mobile platform per the public documentation github.

File: source/examples/getting-started/index2.js [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+const buttonLogin = document.querySelector("#btnLogin");
+const buttonCreate = document.querySelector("#btnCreate");
+buttonCreate.disabled=true;
+const labelCreateStatus = document.querySelector("#labelCreateStatus");
+
+// 1a. Start Login Process. Call session.login() function.
+async function login() {
+  if ( !session.info.isLoggedIn && !new URL(window.location.href).searchParams.get("code")) {
+    await session.login({
+      oidcIssuer: "https://inrupt.net",
```
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/examples/getting-started/index2.js:51]: Done.

File: source/examples/getting-started/index2.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/examples/getting-started/index2.js]: Heh heh. Clearly, I am not either.  I added the blurb.  We'll see if @pmcb55  has strong feelings.

File: source/tutorial/getting-started-part2.rst (hunk 8 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:175]: Heh heh -- I agree but I didn't know if you all would have time for that.   I'll retry and see later this morning.

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:331]: We probably should.  I debated myself.  On one hand, the lazy me was thinking "Well.... we kind of discuss the convenience objects in the getting started part 1 - but in a very light way.  will update.

File: source/tutorial/getting-started-part2.rst (hunk 12 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:342]: Oh -  heh -- just an oversight.  I had used URLs for both and later thought I changed both, but clearly not.   Will update.

File: source/introduction.rst (hunk 4 above)
Comment (Vinnl) [reply source/introduction.rst:14]: I'd think we could at least say "the set of Solid specifications" or "the specifications that make up Solid", but unfortunately I'm only regurgitating something I've seen mentioned before. Not sure if there's someone within Inrupt who can verify this.

(Personally, I don't think it's too big of a deal either way, but a rather large point was made of it last time around. Maybe just @oolivo or @nicolasmondada can chime in how bad it would be it this would potentially not technically be 100% accurate.)

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (Vinnl) [reply source/tutorial/getting-started-part2.rst:331]: I guess for the usage of constants in the first place that might be true, but then why can't we use regular `SCHEMA` - how does Inrupt extend it, and why does it matter? (Assuming that that's what `_INRUPT_EXT` means.)

File: source/examples/getting-started/index2.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (pmcb55) [reply source/examples/getting-started/index2.js]: Hi guys - yeah, Vinnie's is totally correct here, but I agree that just adding some blurb to explain is fine.

Very minor point @kay-kim - we don't capitalise acronyms in our **_code_**, i.e. function names, variables, class names, etc., but we do if writing **_English_**, i.e. comments, or message string, e.g. that code snippet would be:
```
labelCreateStatus.textContent = "";
  const podUrl = document.getElementById("PodURL").value;

  const dsUrl = `${podUrl}/getting-started/readingList/myListDS`
```
Also, _generally_ I dislike acronyms, so for instance I don't know what `DS` is referring to here. If it's just `Dataset`, then it hardly seems warranted to truncate that to `DS`.

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (pmcb55) [reply source/tutorial/getting-started-part2.rst]: Being totally predantic (as always!), but `savedReadListDataSet` shouldn't be capitalising `Set`, so it should be `savedReadListDataset`. Also I assume a `ReadList` is a list of things that I've already Read, as opposed to a `ReadingList`?

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (NSeydoux) [reply source/tutorial/getting-started-part2.rst:331]: `SCHEMA_INRUPT_EXT` is exactly that: an extension of the `schema.org` vocabulary, in particular adding language tags/translations to labels and comments lacking them in the canonical `schema.org`.

**Review by NSeydoux — APPROVED** [review]

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (Vinnl) [reply source/tutorial/getting-started-part2.rst:331]: OK, so in that case... Regular plain old `SCHEMA` should suffice, right, since  we don't need tags/translations (or even labels and comments) here?

Comment (pmcb55) [reply source/tutorial/getting-started-part2.rst:331]: So this 'Inrupt extension' does more than just what Nic says - it also cherry-picks just a handful of terms from the 1,800 or so terms from Schema.org, thereby reducing the bundle size (and auto-complete noise, and just 'noise' in general). I thought there used to be a way to request subsets of terms (as Turtle) from Schema.org (e.g. to just request the Turtle for the Person class and its associated terms), but that doesn't seem to be available anymore.
So TL;DR: I'd just add a simple comment to that explain what SCHEMA_INRUPT_EXT is, and explain that it's an Inrupt extension to make working with select terms from Schema.org easier and quicker (i.e. rather than dealing with the clutter of all 1,800+ terms in there today).

Comment (pmcb55) [reply source/tutorial/getting-started-part2.rst:331]: And by the way, I removed the full-blown SCHEMA generated artifact from the generated bundle of common RDF vocabs. We could add it back easily at any time, or just offer it as it's own standalone bundle, but given the size of that bundle I didn't think anyone would really want to load all 1,800 vocab terms into their full-front apps.

File: source/includes/table-client-libraries.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/includes/table-client-libraries.rst:13]: Thanks much.  Appreciate your thoroughness though on both counts :D

File: source/examples/getting-started/index2.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (kay-kim) [reply source/examples/getting-started/index2.js]: 👍  am updating in the upcoming patch.

File: source/introduction.rst (hunk 4 above)
Comment (kay-kim) [reply source/introduction.rst:14]: 👍  will make it Solid specifications for now in the upcoming patch, and wrangle Oz and NicolasM for a decision and we can make the change one way or the other across all our docs.

File: source/tutorial/getting-started-part2.rst (hunk 9 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:196]: Updated.  Will be in next patch

File: source/tutorial/getting-started-part2.rst (hunk 14 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst]: Pat, Pat, Pat ... Haven't you been in a book club before?  One doesn't actually need to read things; it's enough just saving a list 😉   Heh -- updated.  And am removing  the Dataset suffix to these. Can put them back (with the appropriate spelling, of course).   Am of two minds - on one hand, it's nice to re-enforce the idea of a SolidDataset and then Things, etc.  But, then, the variable names seem so unwieldy.  Ugh.

File: source/tutorial/getting-started-part2.rst (hunk 11 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:331]: Updated.  Will be in next patch.

File: source/tutorial/getting-started-part2.rst (hunk 12 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:342]: Updated.

Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst]: Oh --so, it turns out there isn't SCHEMA_INRUPT_EXT.article (am looking at the installed SCHEMA_INRUPT_EXT.ts)  -- I only did a  grep and just looked whether it returned a match, not at what actually hit -- SCHEMA_INRUPT_EXT has a match for a  comment with the word article in it 😞   It should know what I meant.  ugh. Updated to use ActivityStream vocab.

File: source/tutorial/getting-started-part2.rst [hunk 21]
```diff
@@ -332,28 +332,33 @@ libraries to write data to your Pod.
          :language: javascript
          :lines: 89-95
 
-      ``RDF`` and ``SCHEMA_INRUPT_EXT`` provide convenience objects for
-      many Solid-related terms:
+      ``RDF``, ``SCHEMA_INRUPT_EXT``, and ``AS`` provide convenience
```
Comment (pmcb55) [inline source/tutorial/getting-started-part2.rst:335]: There are repeated references to the vocabs `providing convenience objects`. Maybe you've discussed this before with the guys, but I'd suggest referring to them as `providing convenience constants`, and then this sentence might become `providing convenience constants that represent many of the ideas and concepts that are useful in Solid itself, and in Solid applications`.

File: source/tutorial/getting-started-part2.rst [hunk 22]
```diff
@@ -332,28 +332,33 @@ libraries to write data to your Pod.
...
-        ``http://www.w3.org/1999/02/22-rdf-syntax-ns#type``.
+        ``RDF.type`` is a convenience object that includes the
+        ``http://www.w3.org/1999/02/22-rdf-syntax-ns#type`` IRI.
 
       - ``SCHEMA_INRUPT_EXT`` is Inrupt's extension of the `schema.org
         Vocabulary <http://schema.org>`_. It provides convenience
         objects for a **subset** of terms from the `schema.org
         Vocabulary`_, adding language
         tags/translations to labels and comments if missing from
         schema.org.
```
Comment (pmcb55) [inline source/tutorial/getting-started-part2.rst:348]: I'd add a simple justification for **_why_** it's a subset - i.e. just append a sentence like "Schema.org itself currently defines over 2,500 terms (see [Schema.org](https://schema.org/docs/schemas.html)), whereas most applications (including Solid itself) really only require tiny specialized subsets of those terms. By using SCHEMA_INRUPT_EXT, which cherry-picks a small set of very generally applicable terms, we reduce noise, clutter and bundle sizes - but if you require a Schema.org term not currently cherry-picked by Inrupt in our extension, you can always declare it manually in your own code, create your own extension vocabulary, or request that Inrupt add that term to SCHEMA_INRUPT_EXT."

File: source/tutorial/getting-started-part2.rst (hunk 8 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst]: Removed the sentence.  I no longer get warnings (I think explicitly installing the libraries now instead of having it automatically do it when running did the trick).

File: source/tutorial/getting-started-part2.rst (hunk 21 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst:335]: Heh - I believe convenience objects was actually your suggestion during the using vocabularies review :D 

> provide convenient objects (as JavaScript classes) containing static constants for common identifiers used right across Solid (and RDF in general). Importing these classes simply means developers don't need to hard-code these identifiers themselves in their code...

Heh - I had shortened it.  I'll footnote the details to break up the text.

File: source/tutorial/getting-started-part2.rst (hunk 22 above)
Comment (kay-kim) [reply source/tutorial/getting-started-part2.rst]: Done - put  in the Vocabularies page with a shortened version in the getting started (with a reference to the SCHEMA_INRUPT_EXT) section.

What will probably happen (post-GA) is I'd like to create a reference page with these, so that when talking about FOAF, SCHEMA_INRUPT_EXT, and others as part of an example, we can just add 1-3 sentences in the context and have the readers go to a page that provides a fuller reference on these things.
